models/appModel.js

```diff
@@ -0,0 +1,18 @@
+var AppModel = Backbone.Model.extend({
+  defaults: function() {
+    return{
+      videos: new VideosCollection(),
+      playingVideo: null,
+      query: ''
+    };
+  },
+
+  initialize: function(){
+    this.on('change:query', this.passQueryToCollection);
+    this.listenTo(this.get('videos') ,"reset", this.setCurrentVideo);
+  },
+
+  passQueryToCollection: function(){
+    this.get('videos').fetchNewQuery(this.get('query'));
```

Comment:
I would have preferred to see you set the query on the collection and then have the collection listen for changes and re-query as necessary.